### PR TITLE
chore: add msg and json file types

### PR DESCRIPTION
**Pull request by ryannikolaidis**
`.msg` type doesn't get its mimetype resolved by `mimetypes.guess_type`, because of this we need an extra conditional here.

* add msg file type from filename not handled by mimetypes library
* update tests
* bonus: change test `convert_files_for_api` option from `another_md_mimetype` to `use_octet_stream_type` since we now need this to validate msg type files and may need for others later.

### CODE REVIEW

File: unstructured_api_tools/pipelines/templates/pipeline_api.txt [hunk 1]
```diff
@@ -40,9 +40,12 @@ def get_validated_mimetype(file):
     if not content_type or content_type == "application/octet-stream":
         content_type = mimetypes.guess_type(str(file.filename))[0]
 
-        # Markdown mimetype is too new for the library - just hardcode that one in for now
-        if not content_type and ".md" in file.filename:
-            content_type = "text/markdown"
+        # Some filetypes missing for this library, just hardcode them for now
+        if not content_type:
+            if ".md" in file.filename:
```
Comment (cragwolfe) [inline unstructured_api_tools/pipelines/templates/pipeline_api.txt:45]: is this too lenient a condition compared with say endswith?

**Comment by cragwolfe** [comment]
Should there be another test for an unstructured .json payload?
This is great, by the way. 🙏

File: unstructured_api_tools/pipelines/templates/pipeline_api.txt (hunk 1 above)
Comment (ryannikolaidis) [reply unstructured_api_tools/pipelines/templates/pipeline_api.txt:45]: yes, definitely!

**Comment by ryannikolaidis** [comment]
> Should there be another test for an unstructured .json payload? This is great, by the way. 🙏

yep, I'll add that now

**Comment by cragwolfe** [comment]
@ryannikolaidis ,
correction: could be any .json file payload (does not need to be an "unstructured .json")

**Comment by ryannikolaidis** [comment]
> @ryannikolaidis , correction: could be any .json file payload (does not need to be an "unstructured .json")

ack, just saw this. I did grab from unstructured, but happy to swap something else out.

**Review by cragwolfe — APPROVED** [review]
LGTM!

**Comment by cragwolfe** [comment]
> ack, just saw this. I did grab from unstructured, but happy to swap something else out.

no worries, the unstructured .json is also a json, so that works :)

**Pull request merged**